blockchain/sync_manager/src/sync.rs

```diff
@@ -15,14 +15,16 @@ use libp2p::core::PeerId;
 use message::{Message, MsgMeta};
 use num_bigint::BigUint;
 use raw_block::RawBlock;
-use state::{HamtStateTree, StateTree};
+use state_manager::StateManager;
+use state_tree::{HamtStateTree, StateTree};
 use std::collections::HashMap;
 
+/// Syncer updates the key-value store based on series of validation checks adhering to consensus rules, can query
+/// the network for blocks and assists in informing the network of incoming blocks
 pub struct Syncer<'a> {
     // TODO add ability to send msg to all subscribers indicating incoming blocks
-    // TODO add state manager
     // TODO add block sync
-
+    state_manager: StateManager<'a>,
     // manages sync buckets
     sync_manager: SyncManager<'a>,
```

Comment:
Should assign a different lifetime for these variables (unless they have the same lifetime for sure)